### ISSUE TITLE
Any built-in affordances for detecting database updates? 

### ISSUE THREAD

**Issue by jackpal** [facebookincubator/CG-SQL#100]
I'm trying to use cg-sql to write a toy todo list app.

One of the things that the app needs to do is detect when the query result set needs to be re-fetched due to the tables changing.

One way to do this would be to use sqlite3_update_hook, and detect changes to the tables that are used by the query.

This seems like something that many cg-sql clients would want to do, and it seems like it's possible to read the necessary information from the json schema. (I think it's the query "usesTables" property.)

Is that how things are supposed to work? It requires clients to either parse the the json schema at runtime or write their own code gen on top of the existing code gen.

I guess an alternative would be to have some my_stored_proc_uses_tables API generated in the C code for each stored procedure.

**Comment by ricomariani** [comment]
The JSON schema is designed to help with this.  You can write a little update hook that tells you when your queries are stale using the dependency info there.  A little python to create a C function of your choice to manage the updates is the way to go.  It would likely look a lot like the cqljson.py that I just checked in for other dumps.  Messenger uses this mechanism but the problem is what we do wouldn't be very useful to anyone else.  I don't have a generic answer for this because you have to answer the question:  what should the notification look like for any given app?

So basically wire up something that notifies how you like, the info you need is right there.  The other thing you need to do is some kind of aggregate build that has all the stored procedures in it.  Or else you unify the results of many runs.  Again, these are all choices the compiler is silent on.

**Comment by ricomariani** [comment]
One of the major functions of the JSON is to give you a way to generate whatever glue you might need in your world.  It has basically full API and dependencies for your universe.

**Comment by jackpal** [comment]
Thanks! I'll try that.
